.github/workflows/code-review.yml

```diff
@@ -0,0 +1,64 @@
+name: Code Review
+
+on:
+  workflow_dispatch:
+  pull_request:
+    types: [opened, synchronize]
+    paths:
+      - "*"
+      - "!infra/*"
+      - "!.github/*"
+
+jobs:
+  js_code_review:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Check-out code
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+      # Corepack is an official tool by Node.js that manages package managers versions
+
+      # This is needed to avoid
+      # Error: Error when performing the request to https://registry.npmjs.org/yarn/latest; 
+      - name: Setup target Node.js to enable Corepack
+        uses: actions/setup-node@1e60f620b9541d16bece96c5465dc8ee9832be0b # v4.0.3
+        with:
+          node-version-file: ".node-version"
+
+      - name: Setup yarn
+        run: corepack enable
+
+      - name: Setup Node.js
+        uses: actions/setup-node@1e60f620b9541d16bece96c5465dc8ee9832be0b # v4.0.3
```

Comment:
it looks like setup-node step is called twice

Reply:
The reason for the two-steps is indicated in the comment starting with `Error:`, it is a workaround that ensures the correct setup of Node